poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -443,96 +466,95 @@ public void checkAndUpdateIntegrationStatus(FacilitySoftwareConfig powerIQ, Stri
       return dataCentersMap;
    }
 
-   public Map<String, Asset> getPDUAssetMap() {
-      Map<String, Asset> assetsMap = new HashMap<String, Asset>();
-      try {
-         FacilitySoftwareConfig[] nlytes =
-               restClient.getFacilitySoftwareByType(SoftwareType.Nlyte).getBody();
-         if (nlytes == null || nlytes.length == 0) {
-            return assetsMap;
-         }
-         for (FacilitySoftwareConfig nlyte : nlytes) {
-            List<Asset> nlytePdus =
-                  restClient.getAllAssetsBySourceAndType(nlyte.getId(), AssetCategory.PDU);
-            for (Asset asset : nlytePdus) {
-               assetsMap.put(asset.getAssetName().toLowerCase(), asset);
-            }
-         }
-
-      } catch (Exception e) {
-         logger.error("Failed to get PDU Maps", e);
-      }
-      return assetsMap;
-   }
-
-   public List<Asset> getSensorMetaData(PowerIQAPIClient client, String assetSource) {
-      List<Asset> assets = new ArrayList<Asset>();
-      List<Asset> pdus = new ArrayList<Asset>();
+   public void saveSensorAssetsToFlowgate(Map<String, Asset> exsitingSensorAssets,
+         PowerIQAPIClient client, String assetSource, Map<String, Asset> pduAssetMap) {
       List<Sensor> sensors = getSensors(client);
```

Comment:
for the sensors do we have get sensor by page method?  what i'm thinking is that if the sensor number is huge. we still will run into trouble. since by default the sensor number is double size of pdu.